qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/resolvers/OrValueResolverTest.java

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.resolvers;
+
+import static com.redhat.qute.QuteAssert.assertHover;
+import static com.redhat.qute.QuteAssert.c;
+import static com.redhat.qute.QuteAssert.r;
+import static com.redhat.qute.QuteAssert.testCompletionFor;
+import static com.redhat.qute.QuteAssert.testDiagnosticsFor;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for 'or' value resolver.
+ *
+ * @see https://quarkus.io/guides/qute-reference#built-in-resolvers
+ *
+ */
+public class OrValueResolverTest {
+
+	@Test
+	public void diagnostics() {
+		String template = "{@org.acme.Item item}\r\n" + //
+				"{item.or}";
+		testDiagnosticsFor(template);
+
+		template = "{@org.acme.Item item}\r\n" + //
+				" \r\n" + //
+				"{item.name ?: 'John'}";
+		testDiagnosticsFor(template);
+
+		template = "{@org.acme.Item item}\r\n" + //
+				" \r\n" + //
+				"{item.name.or('John')}";
+		testDiagnosticsFor(template);
+	}
+
+	@Test
+	public void completion() throws Exception {
+		String template = "{@org.acme.Item item}\r\n" + //
+				"{item.name.or.|}";
+		testCompletionFor(template, //
+				c("or(base: T, arg : java.lang.Object) : T", "or", r(1, 20, 1, 20)));
+	}
+
+	@Test
+	public void hover() throws Exception {
+		String template = "{{@org.acme.Item item}\r\n" + //
+				" \r\n" + //
+				"{item.name.o|r('John')}";
+		assertHover(template, "```java" + //
+				System.lineSeparator() + //
+				"Item or()" + //
```

Comment:
replace with `"T or(Object arg)" + //`

for the moment T generic is not resolved.